File: hospitalManagement/Nurses.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hospitalManagement
{
    class Nurses:IBasicActivities<Nurse>
    {
        //field
        private List<Nurse> nurseList;
        private long count;


        // Infor field
        //  Management field
        // Dynamic 
        // Properties

        public long Count { get => count; set => count = value; }
        internal List<Nurse> NurseList { get => nurseList; set => nurseList = value; }

        // Constructors

        public Nurses()
        {
            nurseList = new List<Nurse>();
            count = 0;
        }
        public Nurses(List<Nurse> nurseList, long count)
        {
            this.nurseList = nurseList;
            this.count = count;
        }

        

        public void AddItem()
        {
            Console.WriteLine("Add nurse");

            Nurse nurse = new Nurse();
            nurse.Input();
            nurseList.Add(nurse);
            this.Count++;
            Console.WriteLine("Done!");

        }

        public void AddItem(Nurse value)
        {
            Console.WriteLine("Add nurse");

            nurseList.Add(value);
            this.Count++;
            Console.WriteLine("Done!");

        }

        public bool Clear()
        {
            Console.WriteLine("Clear all nurses");

            nurseList.Clear();
            if (nurseList.Count > 0)
            {
                Console.WriteLine("Failure");
                return false;
            }
            this.Count = 0;
            Console.WriteLine("Done!");
            return true;
        }

        public Nurse FindItem(string id)
        {
            Nurse res = null;
            nurseList.ForEach(value =>
            {
                if (value.Id == id)
                {
                    res = value;
                }
            });
            if (res == null)
            {
                Console.WriteLine($"Not found nurse with id: {id}");

            }
            else
            {
                Console.WriteLine("Done!");

            }
            return res;
        }

        public bool RemoveItem(string id)
        {
            Console.WriteLine("Remove the nurse");

            bool res = false;
            nurseList.ForEach(value =>
            {
                if (value.Id == id)
                {
                    nurseList.Remove(value);
                    res = true;
                    this.Count--;
                }
            });
            if (res == false)
            {
                Console.WriteLine($"Not found nurse with id: {id}");

            }
            else
            {
                Console.WriteLine("Done!");

            }
            return res;
        }

        public void ShowInformation()
        {
            Console.WriteLine("Show information of all nurses");

            nurseList.ForEach(value => { value.Output(); Console.WriteLine(); });
            Console.WriteLine("Done!");
        }

        public Nurse UpdateItem(string id)
        {
            Console.WriteLine("Update the nurse");

            Nurse res = null;
            nurseList.ForEach(value =>
            {
                if (value.Id == id)
                {
                    value.Input();
                    res = value;
                }
            });
            if (res == null)
            {
                Console.WriteLine($"Not found the nurse with id:{id}");

            }
            else
            {
                Console.WriteLine("Done!");

            }
            return res;
        }
    }
}

File: src/hospitalManagement/Nurses.java
package hospitalManagement;

import java.util.ArrayList;
//import java.util.Scanner;
public class Nurses implements IBasicActivities<Nurse>
{
    //field
    private ArrayList<Nurse> nurseList;
    private long count;


    // Information field
    //  Management field
    // Dynamic 
    // Properties

    // Constructors

    public ArrayList<Nurse> getNurseList() {
        return nurseList;
    }
    public void setNurseList(ArrayList<Nurse> nurseList) {
        this.nurseList = nurseList;
    }
    public long getCount() {
        return count;
    }
    public void setCount(long count) {
        this.count = count;
    }
    public Nurses()
    {
        nurseList = new ArrayList<Nurse>();
        count = 0;
    }
    public Nurses(ArrayList<Nurse> nurseList, long count)
    {
        this.nurseList = nurseList;
        this.count = count;
    }

    

    public void AddItem()
    {
        System.out.println("Add nurse");

        Nurse nurse = new Nurse();
        nurse.Input();
        nurseList.add(nurse);
        this.count++;
        System.out.println("Done!");

    }

    public void AddItem(Nurse value)
    {
        System.out.println("Add nurse");

        nurseList.add(value);
        this.count++;
        System.out.println("Done!");

    }

    public boolean Clear()
    {
        System.out.println("Clear all nurses");

        nurseList.clear();
        
        System.out.println("Done!");
        return true;
    }

    public Nurse FindItem(String id)
    {
        Nurse res = null;
        for (Nurse i : nurseList) {
			if( i.getId() == id)
			{
				res = i;
			}
		}
        if (res == null)
        {
            System.out.println("Not found nurse with id: " + id);

        }
        else
        {
            System.out.println("Done!");

        }
        return res;
    }

    public boolean RemoveItem(String id)
    {
        System.out.println("Remove the nurse");

        boolean res = false;
        for (Nurse i : nurseList) {
            if( i.getId() == id)
            {
                nurseList.remove(i);
                this.count--;
            }
        }
        if (res == false)
        {
            System.out.println("Not found nurse with id: " + id);

        }
        else
        {
            System.out.println("Done!");

        }
        return res;
    }

    public void ShowInformation()
    {
        System.out.println("Show information of all nurses");

        nurseList.forEach(value -> { value.Output(); System.out.println(); });
        System.out.println("Done!");
    }

    public Nurse UpdateItem(String id)
    {
        System.out.println("Update the nurse");

        Nurse res = null;
        for (Nurse i : nurseList) {
            if( i.getId() == id)
            {
                i.Input();
                res = i;
            }
        }
        if (res == null)
        {
            System.out.println("Not found the nurse with id: " + id);

        }
        else
        {
            System.out.println("Done!");

        }
        return res;
    }
}